extensions/positron-assistant/src/extension.ts

```diff
@@ -114,26 +114,47 @@ export async function registerModels(context: vscode.ExtensionContext, storage:
 async function registerModelWithAPI(modelConfig: ModelConfig, context: vscode.ExtensionContext, isDefault = false) {
 	// Register with Language Model API
 	if (modelConfig.type === 'chat') {
+		const models = availableModels.get(modelConfig.provider);
+		const modelsCopy = models ? [...models] : [];
+
 		const languageModel = newLanguageModel(modelConfig);
 		const error = await languageModel.resolveConnection(new vscode.CancellationTokenSource().token);
 
 		if (error) {
 			throw new Error(error.message);
 		}
 
-		const modelDisp = vscode.lm.registerChatModelProvider(languageModel.identifier, languageModel, {
-			name: languageModel.name,
-			family: languageModel.provider,
-			vendor: context.extension.packageJSON.publisher,
-			version: context.extension.packageJSON.version,
-			capabilities: languageModel.capabilities,
-			maxInputTokens: 0,
-			maxOutputTokens: 0,
-			isUserSelectable: true,
-			isDefault: isDefault,
-		});
-		modelDisposables.push(modelDisp);
-		vscode.commands.executeCommand('setContext', hasChatModelsContextKey, true);
+		if (modelsCopy.length === 0) {
+			// use the default model
+
+			modelsCopy.push({
+				name: modelConfig.name,
+				identifier: modelConfig.model,
+			});
+		}
+
+		for (const model of modelsCopy) {
+			const newConfig = {
+				...modelConfig,
+				model: model.identifier,
+				name: model.name,
+			};
+			const languageModel = newLanguageModel(newConfig);
+
+			const modelDisp = vscode.lm.registerChatModelProvider(`${languageModel.identifier}-${model.identifier}`, languageModel, {
+				name: languageModel.name,
+				family: languageModel.provider,
+				vendor: context.extension.packageJSON.publisher,
+				version: context.extension.packageJSON.version,
+				capabilities: languageModel.capabilities,
+				maxInputTokens: 0,
+				maxOutputTokens: 0,
+				isUserSelectable: true,
+				isDefault: isDefault,
```

Comment:
Looks like this could set `isDefault` on multiple models -- intentional?

Reply:
Not at all, I'll update it so it registers the first model from the provider with `isDefault` and change it to `false` if needed for the subsequent models.